{
  "url": "https://api.github.com/repos/ChainSafe/lodestar/issues/3947",
  "repository_url": "https://api.github.com/repos/ChainSafe/lodestar",
  "labels_url": "https://api.github.com/repos/ChainSafe/lodestar/issues/3947/labels{/name}",
  "comments_url": "https://api.github.com/repos/ChainSafe/lodestar/issues/3947/comments",
  "events_url": "https://api.github.com/repos/ChainSafe/lodestar/issues/3947/events",
  "html_url": "https://github.com/ChainSafe/lodestar/issues/3947",
  "id": 1213264827,
  "node_id": "I_kwDOCD5_Gc5IUPO7",
  "number": 3947,
  "title": "Mainnet shadow fork 2 - empty deposit tree ETH1_ERROR_WRONG_DEPOSIT_ROOT",
  "user": {
    "login": "dapplion",
    "id": 35266934,
    "node_id": "MDQ6VXNlcjM1MjY2OTM0",
    "avatar_url": "https://avatars.githubusercontent.com/u/35266934?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/dapplion",
    "html_url": "https://github.com/dapplion",
    "followers_url": "https://api.github.com/users/dapplion/followers",
    "following_url": "https://api.github.com/users/dapplion/following{/other_user}",
    "gists_url": "https://api.github.com/users/dapplion/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/dapplion/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/dapplion/subscriptions",
    "organizations_url": "https://api.github.com/users/dapplion/orgs",
    "repos_url": "https://api.github.com/users/dapplion/repos",
    "events_url": "https://api.github.com/users/dapplion/events{/privacy}",
    "received_events_url": "https://api.github.com/users/dapplion/received_events",
    "type": "User",
    "site_admin": false
  },
  "labels": [
    {
      "id": 1200090446,
      "node_id": "MDU6TGFiZWwxMjAwMDkwNDQ2",
      "url": "https://api.github.com/repos/ChainSafe/lodestar/labels/prio-medium",
      "name": "prio-medium",
      "color": "fdc09f",
      "default": false,
      "description": "Resolve this some time soon (tm)."
    },
    {
      "id": 2159934357,
      "node_id": "MDU6TGFiZWwyMTU5OTM0MzU3",
      "url": "https://api.github.com/repos/ChainSafe/lodestar/labels/scope-testnet-debugging",
      "name": "scope-testnet-debugging",
      "color": "980043",
      "default": false,
      "description": "Issues uncovered through running a node on a public testnet."
    }
  ],
  "state": "closed",
  "locked": false,
  "assignee": null,
  "assignees": [

  ],
  "milestone": null,
  "comments": 4,
  "created_at": "2022-04-23T10:23:36Z",
  "updated_at": "2023-05-26T17:44:10Z",
  "closed_at": "2023-05-26T17:44:09Z",
  "author_association": "MEMBER",
  "active_lock_reason": null,
  "body": "<!--NOTE: -->\r\n<!--- General questions should go to the discord chat instead of the issue tracker.-->\r\n\r\n**Describe the bug**\r\n\r\nShadow fork 2 lodestar geth combination (only checked this one) failed to propose all blocks between genesis and genesis +3 hours\r\n\r\n```\r\nApr-22 09:58:48.067[API]             error: Req req-1f2 produceBlock error  code=ETH1_ERROR_WRONG_DEPOSIT_ROOT, root=0xff836677895bc3325cfcc752ff966cda93fd773ee6e602b18977630d5e527932, expectedRoot=0xd70a234731285c6804c2a4f56711ddb8c82c99740f207854891028af34e27e5e\r\nError: ETH1_ERROR_WRONG_DEPOSIT_ROOT\r\n    at getDepositsWithProofs (/usr/app/node_modules/@chainsafe/lodestar/src/eth1/utils/deposits.ts:52:11)\r\n    at Eth1DepositsCache.get (/usr/app/node_modules/@chainsafe/lodestar/src/eth1/eth1DepositsCache.ts:34:33)\r\n    at getDeposits (/usr/app/node_modules/@chainsafe/lodestar/src/eth1/utils/deposits.ts:30:20)\r\n    at Eth1DepositDataTracker.getDeposits (/usr/app/node_modules/@chainsafe/lodestar/src/eth1/eth1DepositDataTracker.ts:116:12)\r\n    at Eth1DepositDataTracker.getEth1DataAndDeposits (/usr/app/node_modules/@chainsafe/lodestar/src/eth1/eth1DepositDataTracker.ts:77:22)\r\n    at assembleBody (/usr/app/node_modules/@chainsafe/lodestar/src/chain/factory/block/body.ts:64:32)\r\n    at assembleBlock (/usr/app/node_modules/@chainsafe/lodestar/src/chain/factory/block/index.ts:43:11)\r\n    at Object.produceBlock (/usr/app/node_modules/@chainsafe/lodestar/src/api/impl/validator/index.ts:180:21)\r\n    at Object.handler (/usr/app/node_modules/@chainsafe/lodestar-api/src/server/utils/server.ts:61:23)\r\n```\r\n\r\nat genesis +3 hours the node restarted and could propose 100% of blocks.\r\n\r\nNote that the deposit queue is empty during the first 4 hours because there has not been a vote yet on eth1 data. Could it be some issue handling an empty deposit tree?\r\n\r\n**Expected behavior**\r\n\r\nPropose all blocks\r\n",
  "closed_by": {
    "login": "dapplion",
    "id": 35266934,
    "node_id": "MDQ6VXNlcjM1MjY2OTM0",
    "avatar_url": "https://avatars.githubusercontent.com/u/35266934?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/dapplion",
    "html_url": "https://github.com/dapplion",
    "followers_url": "https://api.github.com/users/dapplion/followers",
    "following_url": "https://api.github.com/users/dapplion/following{/other_user}",
    "gists_url": "https://api.github.com/users/dapplion/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/dapplion/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/dapplion/subscriptions",
    "organizations_url": "https://api.github.com/users/dapplion/orgs",
    "repos_url": "https://api.github.com/users/dapplion/repos",
    "events_url": "https://api.github.com/users/dapplion/events{/privacy}",
    "received_events_url": "https://api.github.com/users/dapplion/received_events",
    "type": "User",
    "site_admin": false
  },
  "reactions": {
    "url": "https://api.github.com/repos/ChainSafe/lodestar/issues/3947/reactions",
    "total_count": 0,
    "+1": 0,
    "-1": 0,
    "laugh": 0,
    "hooray": 0,
    "confused": 0,
    "heart": 0,
    "rocket": 0,
    "eyes": 0
  },
  "timeline_url": "https://api.github.com/repos/ChainSafe/lodestar/issues/3947/timeline",
  "performed_via_github_app": null,
  "state_reason": "completed"
}
[
  {
    "url": "https://api.github.com/repos/ChainSafe/lodestar/issues/comments/1253151983",
    "html_url": "https://github.com/ChainSafe/lodestar/issues/3947#issuecomment-1253151983",
    "issue_url": "https://api.github.com/repos/ChainSafe/lodestar/issues/3947",
    "id": 1253151983,
    "node_id": "IC_kwDOCD5_Gc5KsZTv",
    "user": {
      "login": "stale[bot]",
      "id": 26384082,
      "node_id": "MDM6Qm90MjYzODQwODI=",
      "avatar_url": "https://avatars.githubusercontent.com/in/1724?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/stale%5Bbot%5D",
      "html_url": "https://github.com/apps/stale",
      "followers_url": "https://api.github.com/users/stale%5Bbot%5D/followers",
      "following_url": "https://api.github.com/users/stale%5Bbot%5D/following{/other_user}",
      "gists_url": "https://api.github.com/users/stale%5Bbot%5D/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/stale%5Bbot%5D/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/stale%5Bbot%5D/subscriptions",
      "organizations_url": "https://api.github.com/users/stale%5Bbot%5D/orgs",
      "repos_url": "https://api.github.com/users/stale%5Bbot%5D/repos",
      "events_url": "https://api.github.com/users/stale%5Bbot%5D/events{/privacy}",
      "received_events_url": "https://api.github.com/users/stale%5Bbot%5D/received_events",
      "type": "Bot",
      "site_admin": false
    },
    "created_at": "2022-09-21T03:12:48Z",
    "updated_at": "2022-09-21T03:12:48Z",
    "author_association": "NONE",
    "body": "This issue has been automatically marked as stale because it has not had recent activity. It will be closed in 15 days if no further activity occurs. Thank you for your contributions.\n",
    "reactions": {
      "url": "https://api.github.com/repos/ChainSafe/lodestar/issues/comments/1253151983/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/ChainSafe/lodestar/issues/comments/1255872483",
    "html_url": "https://github.com/ChainSafe/lodestar/issues/3947#issuecomment-1255872483",
    "issue_url": "https://api.github.com/repos/ChainSafe/lodestar/issues/3947",
    "id": 1255872483,
    "node_id": "IC_kwDOCD5_Gc5K2xfj",
    "user": {
      "login": "philknows",
      "id": 58080811,
      "node_id": "MDQ6VXNlcjU4MDgwODEx",
      "avatar_url": "https://avatars.githubusercontent.com/u/58080811?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/philknows",
      "html_url": "https://github.com/philknows",
      "followers_url": "https://api.github.com/users/philknows/followers",
      "following_url": "https://api.github.com/users/philknows/following{/other_user}",
      "gists_url": "https://api.github.com/users/philknows/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/philknows/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/philknows/subscriptions",
      "organizations_url": "https://api.github.com/users/philknows/orgs",
      "repos_url": "https://api.github.com/users/philknows/repos",
      "events_url": "https://api.github.com/users/philknows/events{/privacy}",
      "received_events_url": "https://api.github.com/users/philknows/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2022-09-23T07:17:40Z",
    "updated_at": "2022-09-23T07:17:40Z",
    "author_association": "MEMBER",
    "body": "Are we still experiencing this?",
    "reactions": {
      "url": "https://api.github.com/repos/ChainSafe/lodestar/issues/comments/1255872483/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/ChainSafe/lodestar/issues/comments/1261972909",
    "html_url": "https://github.com/ChainSafe/lodestar/issues/3947#issuecomment-1261972909",
    "issue_url": "https://api.github.com/repos/ChainSafe/lodestar/issues/3947",
    "id": 1261972909,
    "node_id": "IC_kwDOCD5_Gc5LOC2t",
    "user": {
      "login": "dapplion",
      "id": 35266934,
      "node_id": "MDQ6VXNlcjM1MjY2OTM0",
      "avatar_url": "https://avatars.githubusercontent.com/u/35266934?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/dapplion",
      "html_url": "https://github.com/dapplion",
      "followers_url": "https://api.github.com/users/dapplion/followers",
      "following_url": "https://api.github.com/users/dapplion/following{/other_user}",
      "gists_url": "https://api.github.com/users/dapplion/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/dapplion/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/dapplion/subscriptions",
      "organizations_url": "https://api.github.com/users/dapplion/orgs",
      "repos_url": "https://api.github.com/users/dapplion/repos",
      "events_url": "https://api.github.com/users/dapplion/events{/privacy}",
      "received_events_url": "https://api.github.com/users/dapplion/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2022-09-29T08:52:28Z",
    "updated_at": "2022-09-29T08:52:28Z",
    "author_association": "MEMBER",
    "body": "> Are we still experiencing this?\r\n\r\nThe only way to experience this is with a shadow-fork that has an empty deposit contract for more than 1 eth1 voting period. This condition doesn't happen on any existing network. We may choose to put this issue on ice until shadow forks happen again for capella",
    "reactions": {
      "url": "https://api.github.com/repos/ChainSafe/lodestar/issues/comments/1261972909/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/ChainSafe/lodestar/issues/comments/1564719932",
    "html_url": "https://github.com/ChainSafe/lodestar/issues/3947#issuecomment-1564719932",
    "issue_url": "https://api.github.com/repos/ChainSafe/lodestar/issues/3947",
    "id": 1564719932,
    "node_id": "IC_kwDOCD5_Gc5dQ7s8",
    "user": {
      "login": "dapplion",
      "id": 35266934,
      "node_id": "MDQ6VXNlcjM1MjY2OTM0",
      "avatar_url": "https://avatars.githubusercontent.com/u/35266934?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/dapplion",
      "html_url": "https://github.com/dapplion",
      "followers_url": "https://api.github.com/users/dapplion/followers",
      "following_url": "https://api.github.com/users/dapplion/following{/other_user}",
      "gists_url": "https://api.github.com/users/dapplion/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/dapplion/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/dapplion/subscriptions",
      "organizations_url": "https://api.github.com/users/dapplion/orgs",
      "repos_url": "https://api.github.com/users/dapplion/repos",
      "events_url": "https://api.github.com/users/dapplion/events{/privacy}",
      "received_events_url": "https://api.github.com/users/dapplion/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2023-05-26T17:44:09Z",
    "updated_at": "2023-05-26T17:44:09Z",
    "author_association": "MEMBER",
    "body": "Closing issues for specific incidents on old versions of Lodestar",
    "reactions": {
      "url": "https://api.github.com/repos/ChainSafe/lodestar/issues/comments/1564719932/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  }
]
